src/python/dxpy/scripts/dx.py

```diff
@@ -2516,19 +2516,17 @@ def uninstall(args):
     if app_desc is None:
         try:
             dxpy.api.app_uninstall(args.app)
-            print('Uninstalled application with id: ' + args.app)
+            print('Uninstalled the {id} app'.format(id=args.app))
         except:
-            try:
-                dxpy.api.app_uninstall('app-' + args.app)
-                print('Uninstalled application with name: ' + args.app)
-            except:
-                err_exit()
+            user_data = dxpy.api.user_describe(dxpy.whoami(), {'appsInstalled': True})
```

Comment:
Please front-load the /user-x/describe invocation before attempting any uninstall. Once we verify that either `args.app` or `"app-" + args.app` is in the `appsInstalled` field of the describe response, then we can proceed with the uninstall via /app-x/uninstall (note that this is using app-x, whether that is from `args.app` or from `"app-" + args.app`).